sklearn/metrics/_ranking.py

```diff
@@ -1419,3 +1419,88 @@ def ndcg_score(y_true, y_score, k=None, sample_weight=None, ignore_ties=False):
     _check_dcg_target_type(y_true)
     gain = _ndcg_sample_scores(y_true, y_score, k=k, ignore_ties=ignore_ties)
     return np.average(gain, weights=sample_weight)
+
+
+def top_k_accuracy_score(y_true, y_score, k=5, normalize=True):
+    """Top k Accuracy classification score.
+
+    This metric computes the number of times where the correct label is among
+    the top ``k`` labels predicted (ranked by predicted scores). Note that
+    multilabel classification case isn't handled here.
+
+    Parameters
+    ----------
+    y_true : array-like of shape (n_samples,)
+        True labels.
+
+    y_score : array-like of shape (n_samples, n_classes)
+        Target scores.
+
+    k : int, optional (default=5)
+        Number of guesses allowed to find the correct label.
+
+    normalize : bool, optional (default=True)
+        If ``True``, return the fraction of correctly classified samples.
+        Otherwise, return the number of correctly classified samples.
+
+    Returns
+    -------
+    score : float
+        The best performance is 1 with ``normalize == True`` and the number
+        of samples with ``normalize == False``.
+
+    See also
+    --------
+    accuracy_score
+
+    Notes
+    -----
+    If ``k = 1``, the result will be the same as the accuracy_score (though see
+    note below). If ``k`` is the same as the number of classes, this score will
+    be perfect and meaningless.
+
+    In cases where two or more labels are assigned equal probabilities, the
+    result may be incorrect if one of those labels falls at the threshold, as
+    one class must be chosen to be the ``k``th class and the class chosen may
+    not be the correct one.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.metrics import top_k_accuracy_score
+    >>> y_true = np.array([0, 1, 2, 2])
+    >>> y_score = np.array([[0.5, 0.2, 0.2],
+    ...                     [0.3, 0.4, 0.2],
+    ...                     [0.2, 0.4, 0.3],
+    ...                     [0.7, 0.2, 0.1]])
+    >>> top_k_accuracy_score(y_true, y_score, k=1)
+    0.5
+    >>> top_k_accuracy_score(y_true, y_score, k=2)
+    0.75
+    >>> top_k_accuracy_score(y_true, y_score, k=2, normalize=False)
+    3
+    >>> top_k_accuracy_score(y_true, y_score, k=3)
+    1.0
+
+    """
+    check_consistent_length(y_true, y_score)
+    y_type = type_of_target(y_true)
+
+    if y_type != 'multiclass':
+        raise ValueError(f"Target type must be 'multiclass' not {y_type}")
+
+    y_true = column_or_1d(y_true)
+    y_score = check_array(y_score)
+    classes = _encode(y_true)
+
+    if len(classes) != y_score.shape[1]:
```

Comment:
Since all classes might not be represented in the samples, this should probably check `y_true.max()` against `y_score.shape[1]` instead.

Reply:
Done.